### PR TITLE
[FIX]  Refactoring  update_reseration_event

### PR DESCRIPTION
**Pull request by jananjoy**

### CODE REVIEW

File: project_event/models/project_task.py [hunk 1]
```diff
@@ -413,34 +413,54 @@ def update_reservation_event(self, vals):
...
+        update_vals = {}
+        for index in range(0, len(field_names)):
+            if field_names[index] in vals:
+                if field_names[index] == 'date_start':
+                    update_vals['start'] = vals[field_names[index]]
+                if field_names[index] == 'date_end':
+                    update_vals['stop'] = vals[field_names[index]]
+                if field_names[index] == 'equipment_id' and vals['equipment_id']:
+                    update_vals.update(self.update_value_equipment_id(vals))
+                elif field_names[index] == 'room_id' and vals['room_id']:
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: this is not the equivalent of this code : elif 'room_id' in vals:
                if vals['room_id']:
                    update_vals['equipment_ids'] = \
                        [(6, 0, self.env['resource.calendar.room']
                          .browse(vals['room_id']).instruments_ids.ids)]
                update_vals['room_id'] = vals['room_id']

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/models/project_task.py (hunk 1 above)
Comment (jananjoy) [reply project_event/models/project_task.py]: @rimbendhaou Done

File: project_event/models/project_task.py [hunk 2]
```diff
@@ -427,34 +427,55 @@ def update_reservation_event(self, vals):
         if self.reservation_event_id:
             reservation_event = self.env['calendar.event'].browse(
                 self.reservation_event_id)
-            update_vals = {}
```
Comment (rimbendhaou) [inline project_event/models/project_task.py:430]: Add the unit tests for this

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/models/project_task.py (hunk 2 above)
Comment (jananjoy) [reply project_event/models/project_task.py:430]: @rimbendhaou : Created a new task for writing test cases for this task. You can merge this code.

Comment (rimbendhaou) [reply project_event/models/project_task.py:430]: where?

**Review by rimbendhaou — CHANGES_REQUESTED** [review]
Better to add the unit test in the same PR to be sure that everything works

File: project_event/models/project_task.py (hunk 2 above)
Comment (jananjoy) [reply project_event/models/project_task.py:430]: @rimbendhaou : check in the pending pull request in the list (related with Refactoring update_reseration_event)

**Review by rimbendhaou — APPROVED** [review]

**Pull request merged**